File: booking-app/src/app/models/room-reservation.ts
import { User } from './user';
import { Room } from './room';

export class RoomReservation {
    public Id: number;
    public StartDate: Date;
    public EndDate: Date;
    public TimeStamp: Date;
    public AppUserId: number;
    public AppUser: User ;
    public RoomId: number;
    public Room: Room;

    constructor(){}
}

File: BookingApp/Models/Data/RoomReservation.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Web;

namespace BookingApp.Models.Data
{
    public class RoomReservation
    {
        public int Id { get; set; }

        [Required]
        public DateTime StartDate { get; set; }

        [Required]
        public DateTime EndDate { get; set; }

        [Required]
        public DateTime TimeStamp { get; set; }

        [Required]
        [ForeignKey("Room")]
        public int RoomId { get; set; }

        public Room Room { get; set; }

        [Required]
        [ForeignKey("AppUser")]
        public int AppUserId { get; set; }

        public AppUser AppUser { get; set; }

    }
}